Symfony/CS/Fixer/PSR2/OneLineMethodArgumentFixer.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\PSR2;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Token;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Konrad Cerny <info@konradcerny.cz>
+ */
+class OneLineMethodArgumentFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound('(');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, Tokens $tokens)
+    {
+        /** @var Token $token */
+        foreach ($tokens as $index => $token) {
```

Comment:
```php
<?php

$a = [0, 1];
foreach ($a as $i => $v) {
    var_dump($i);
    $a[] = 'new';
}

var_dump($a);
```

Here, you never iterate through indexes 2 and 3 (ones with `new` value).

Same problem is in your code. You adding new Tokens inside collection. If you add eg 100 tokens then you will ignore last 100 tokens of code, and won't fix it even if they need to.